rdr_service/resource/generators/participant.py

```diff
@@ -1829,6 +1839,47 @@ def _calculate_ubr(self, p_id, summary, ro_session):
 
         return data
 
+    @staticmethod
+    def get_pdf_validation_status(qr_id, consent_authored, ro_session=None):
+        """
+        Determines consent status based on PDF validation status. Initially only applies to EHR PDF validation
+        :param qr_id:  A questionnaire_response_id value
+        :param consent_authored: An authored date for the consent
+        :param ro_session:  A read-only session object
+        """
+        # Cutoff for avoiding incorrectly assuming SUBMITTED_NOT_VALIDATED.  Based on expected release date
+        # of RDR changes that will begin using SUBMITTED_NOT_VALIDATED/SUBMITTED_INVALID statuses.
+        pdf_validation_start_date = datetime.datetime(2023, 3, 10)
+        status = BQModuleStatusEnum.SUBMITTED  # Default, this method is only called for "Yes" consents that have PDF
+        dao = None
+        if not ro_session:
+            dao = ResourceDataDao(backup=True)
+            ro_session = dao.session()
+
+        # Find consent validation result associated with this questionnaire response.  Note, many older consent
+        # validation results do not have a consent_response record because that relation/table was added later
+        sql = """
+             select cf.created, cf.sync_status
+             from consent_response cr
+             join consent_file cf on cr.id = cf.consent_response_id
+             where cr.questionnaire_response_id = :qr_id
+
+        """
```

Comment:
Peggy, can there be more than one 'consent_response' record per questionnaire_response_id?  If so, should there be an 'order by xxxx desc' clause in the sql to get the most recent record?

Reply:
I guess this is possible in normal circumstances for other consents that are not yet part of this PDF validation  logic (EHR specific for now) -- e.g., a Primary consent and a CABOR will have separate `consent_file` validation records / `consent_response` records but have the same `questionnaire_response_id`

There is also what looks like one case of unintended duplication in the `consent_file` table for one EHR consent, where four validation records were created for the same consent (all details in the `consent_file` record match), and four `consent_response` records for that `questionnaire_response_id` were created as well.  

I'll work on an update.